src/structure/structure.ts

```diff
@@ -578,15 +564,15 @@ export class StructureViewer {
     private _removeWidget(guid: string, force: boolean = false) {
 
       if ( this._guids.length > 1 || force === true) {
-        const widgetRoot = document.getElementById(guid);
+        const widgetRoot = getByID(guid) as HTMLElement;
 
-        this.onselect(this._widgetMap[guid].current, `${guid}-nuke`);
+        this.onselect({structure: -1, environment: -1}, `${guid}`);
 
-        if ( widgetRoot !== null) {
-            const oldRoot = widgetRoot.parentNode;
-            if ( oldRoot !== null) {oldRoot.removeChild(widgetRoot); }
+        if ( widgetRoot.parentNode !== null) {
+          widgetRoot.parentNode.removeChild(widgetRoot);
         }
-        const deadCell = document.getElementById(`gi-${guid}`);
+
+        const deadCell = getByID(`gi-${guid}`) as HTMLElement;
```

Comment:
FYI `getByID` returns an HTMLElement by default, and it can return something else by calling it as `getById<HTMLInputElement>(...)`